src/auth/auth.service.ts

```diff
@@ -36,7 +36,28 @@ export class AuthService {
       throw new NotFoundException();
     }
 
-    return await this.jwtService.signAsync({ id: user.id, email: user.email });
+    return await this.jwtService.signAsync(
+      {
+        id: user.id,
+        email: user.email,
+      },
+      { expiresIn: `${process.env.JWT_EXPIRY_TIME}s` },
+    );
+  }
+
+  async createPAT(userId: number): Promise<string> {
+    const validPatToken = await this.userRepository.getUserValidPatWithId(
+      userId,
+    );
+    if (validPatToken) {
+      this.userRepository.invalidateUserPat(userId);
+    }
+    const token = await this.jwtService.signAsync({
+      id: userId,
+      type: 'PAT',
+    });
+    const patToken = await this.userRepository.createPat(userId, token);
+    return patToken.token;
```

Comment:
```suggestion
    const { token } = await this.userRepository.createPat(userId, token);
    return token;
```

I think it's more clear. `patToken.token` sounded strange

Reply:
I'd rename `createPat` to `createPersonalAccessToken` or `generatePersonalAccessToken`

Reply:
Regarding suggestion - sure, but I'd remove `ToDb` part.

But now I'm thinking - why do we have `createPat` method on `userRepository`? Shouldn't we call `personalAccessTokenRepository` instead?